File: ruby/raindrops/raindrops.rb
class Raindrops

  def self.factors?(a, b)
    a % b == 0
  end

  def self.convert(n)
    rain_speak = ""

    rain_speak += 'Pling' if self.factors?(n, 3)
    rain_speak += 'Plang' if self.factors?(n, 5)
    rain_speak += 'Plong' if self.factors?(n, 7)

    unless self.factors?(n, 3) || self.factors?(n, 5) || self.factors?(n, 7)
      rain_speak = n.to_s
    end

    return rain_speak
  end
end

File: csharp/raindrops/Raindrops.cs
using System;

public static class Raindrops
{
    public static string Convert(int number)
    {
        var rain_speak = "";

        if (number % 3 == 0)
        {
            rain_speak += "Pling";
        }
        if (number % 5 == 0)
        {
            rain_speak += "Plang";
        }
        if (number % 7 == 0)
        {
            rain_speak += "Plong";
        }
        if ((number % 3 != 0) && (number % 5 != 0) && (number % 7 != 0))
        {
            rain_speak = number.ToString();
        }

        return rain_speak;
    }
}
